### DISCUSSION TITLE
What's the meaning of i and e in mapping parameters?

### DISCUSSION THREAD

**Discussion by zanglg** [hrsh7th/nvim-cmp#861] [Q&A] [answered]
I notice it in every mapping, but what's the meaning of them? And what I must notice about "i" and "e" if I want to change the default mapping?

**Comment by smjonas** [comment, marked as answer]
`i` is an **I**nsert mode mapping, `c` is for command line mode and `s` for select mode. For example, if you don't use `cmp-cmdline` you should not remap it and only specify `i`. If you are using a snippet engine like UltiSnips that supports select-mode mappings you should also add `s` to the table. Also see the [Wiki](https://github.com/hrsh7th/nvim-cmp/wiki/Example-mappings) for examples :)

**Reply by zanglg** [reply]
Thanks, I see.
